### PR TITLE
Add new API Version 2024-06-01-preview for Microsoft.Resources\changes

### PR DESCRIPTION
**Pull request by manu4rhyme**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [X] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information
PR on private branch already approved to merge
https://github.com/Azure/azure-rest-api-specs-pr/pull/18184

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by manu4rhyme** [comment]
Changes in this PR were already approved in another PR in the private repo here
https://github.com/Azure/azure-rest-api-specs-pr/pull/18184

No breaking change approval is required, since we're adding a new API version

**Comment by mentat9** [comment]
> Changes in this PR were already approved in another PR in the private repo here [Azure/azure-rest-api-specs-pr#18184](https://github.com/Azure/azure-rest-api-specs-pr/pull/18184)
> 
> No breaking change approval is required, since we're adding a new API version

@manu4rhyme - Actually, the BreakingChangeReviewRequired flag is legit. This API version is removing a couple of properties ("propertyChangeType", "changeCategory") from the ChangeBase type which is part of the Microsoft.Resources/changes resource type already published in 2023-07-01-preview, so you need to get a breaking change review for this PR.

Once you have breaking change approval (you can request it in email), I can sign off for ARM. I have confirmed these changes were already signed off by ARM in the previous PR.

**Comment by manu4rhyme** [comment]
Changes in this PR were already approved in another PR in the private repo here
https://github.com/Azure/azure-rest-api-specs-pr/pull/18184

**Comment by ramoka178** [comment]
This cannot be merged, without being reviewed first. Adding it to the queue.

**Comment by ramoka178** [comment]
Signing off based on the previous reviewer's comment + breaking change approval label added

File: specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/changes.json
Comment (TimLovellSmith) [inline specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/changes.json:341]: What if the value before wasn't a string, but an integer? How do you distinguish between integer value of 4, and string value of "4"? Null and "null"? Or just not defined, and null? (which might be represented differently for some resources!) How do you represent array changes?

File: specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/examples/ListChanges.json
Comment (TimLovellSmith) [inline specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/examples/ListChanges.json:40]: Would be great if your examples covered all the types that JSON can represent.

File: specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/examples/GetChange.json
Comment (manu4rhyme) [reply specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/examples/GetChange.json]: Good catch! this should have been just null, updated it.
If it's value is a string "null" then it means the RP for which this change record is encoding null in a string, revealing an issue on RP side, and not Microsoft.Resources/changes

File: specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/changes.json
Comment (manu4rhyme) [reply specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/changes.json:341]: it is an optional/nullable string type, other than null all values including integers/json/xml etc.  are converted to string

File: specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/examples/ListChanges.json
Comment (manu4rhyme) [reply specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/examples/ListChanges.json]: Added example for an array elements value being changed

File: specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/changes.json
Comment (TimLovellSmith) [reply specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/changes.json:341]: Wouldn't that be quite inconvenient though, to consume programmatically, if you were trying to consume these 'changes' and do something like e.g. automatically replicate them on another resource, or roll them back?

You'd have to get the string AND also know the underlying type of the property you're trying to update.

Comment (manu4rhyme) [reply specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/changes.json:341]: Yes it will need an extra step to parse the strings first for now, as it has been since quite a while.
We will consider your feedback in future iterations as it will be a breaking change. For now, this isn't       a blocking comment, so can we go ahead with merge (based on previous review, spanning now over a month)

Comment (TimLovellSmith) [reply specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/changes.json:341]: If its intentionally nullable, I believe you need to annotate it as "x-nullable": true to indicate that null is a legal value.

This is a breaking change.

File: specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/changes.json
Comment (TimLovellSmith) [inline specification/resources/resource-manager/Microsoft.Resources/preview/2024-06-01-preview/changes.json:345]: This would have to be x-nullable true also if it supports nulls

**Review by TimLovellSmith — APPROVED** [review]
LGTM

**Pull request merged**
